### ISSUE TITLE
As a user viewing search results, when my search terms occur in the description I want to see keywords in context so that I can see why the document was included in the search results.

### ISSUE THREAD

**Issue by rlskoeser** [Princeton-CDH/geniza#291]
## testing notes
Go to public search on the test site https://test-geniza.cdh.princeton.edu/documents/ and search for keyword terms that will match description text.
- [x] when search term matches document description, results display search term highlighted in context
- [x] if search term does not match description, you should see the description excerpt that you normally see (you could search on a tag and scroll down to see documents without that term in the description)

Feel free to explore different search options and see how it works:
- multiple search terms 
- wildcards, e.g. avr*
- exact phrase; use quotes around the phrase, e.g. "abu zikri"
- proximity search, e.g. "Moshe Cairo"~5

## dev notes
- [x] make sure document description is indexed in a way to support keywords in context
- [x] document in deploy notes if reindexing is required
- [x] enable highlighting on description in solr query in the document search view (see [example in ppa code](https://github.com/Princeton-CDH/ppa-django/blob/main/ppa/archive/views.py#L182-L187))
- [x] include highlighting results in template context (if needed)
- [x] display description snippet with highlights if there is a match; fall back to existing description display if not
- [x] add scss to highlight matches  — use `apply-theme` scss mixin with the primary theme color
- [x] add/update unit tests for view and template changes

**Comment by richmanrachel** [comment]
@rlskoeser - Descriptions are not coming up (even when they exist) if the search term is not in the description:
![image](https://user-images.githubusercontent.com/68028846/140532234-272c74ec-e019-4ef6-abdd-0faa5bf0fa79.png)

Also @gissoo - as a design note, I don't like that the tags appear above the header for the document. It makes it seem like they're part of the previous document, so it's confusing.

**Comment by richmanrachel** [comment]
Overall the search does seem to work, though! I've never used proximity search before, and it is awesome.

**Comment by rlskoeser** [comment]
Good catch @richmanrachel ! I had implemented the check incorrectly for documents without keyword matches.  Please test again.

**Comment by richmanrachel** [comment]
@rlskoeser - Success! Closing :)

**Comment by richmanrachel** [comment]
@rlskoeser - just leaving a note here (but happy to move it later) from showing Marina this issue:
- Can we make it possible to search specific fields? For example, just in the description? (Right now it's pulling from scholarship info as well).

**Comment by rlskoeser** [comment]
@richmanrachel yes, I've been thinking about that as well. Not for MVP, but we should think about how to approach.

How do you and @mrustow feel about the advanced search syntax examples we implemented on the Prosody Archive? See the examples at the bottom of this page: https://prosody.princeton.edu/search/

**Comment by gissoo** [comment]
> Also @gissoo - as a design note, I don't like that the tags appear above the header for the document. It makes it seem like they're part of the previous document, so it's confusing.
>

> that makes sense! We reached the decision to make this layout pretty much match the cluster search results layout – so I think it would be great if we can bring the tags closer to the title than they are now – @rlskoeser can we make the spacing between the tags and the doc title 4px? I think it's 16px right now

**Comment by rlskoeser** [comment]
That's easy enough on our side. @richmanrachel will this address your concern?

**Comment by richmanrachel** [comment]
@rlskoeser and @gissoo - I still think having the header be on top of all the information that it's referring to would be best.

**Comment by gissoo** [comment]
@richmanrachel @rlskoeser in that case we might want to keep this as is for the cluster interface and change it only for the regular search interface – because it's important to have the tags stay on top for the cluster search. 
would that work? 

If it works, then @richmanrachel please feel free to create an issue for me to revise the design so we don't lose track of this

**Comment by gissoo** [comment]
I would like to mention that that the type size for the tags is pretty small and I'm thinking to apply a grey color to it once we add colors – so it should not be very prominent at all, but let me know about my comment, above.

**Comment by richmanrachel** [comment]
@gissoo - I think I'll need to see it with the colors to properly visualize it. But I'm still hesitant to cede the top space to anything but the title of the document. We can discuss it more at the design meeting?

**Comment by gissoo** [comment]
@richmanrachel sure, discussing it maybe easier!

**Comment by rlskoeser** [comment]
@gissoo : I like it visually, but I should mention that in the html we actually don't follow this order because I didn't think it would make sense for a screen reader to get tags before they get a label for a document.  It would be better if the display matched the document order (especially once we make the title clickable, since if we do not tab order will not match display order).

Given that the tags serve a different purpose here than we expect they will for the cluster search (based on the prototype, where tag-pairs are used to describe a _group_ of results), I think we should reorder so tags come after titles.

**Comment by gissoo** [comment]
@rlskoeser thanks for writing, I wondered what your answer to my question above would be. Yes, then you're ok with keeping it as is for the cluster search – that makes sense to me too – For this search it makes a lot of sense, but for the cluster search the tags/terms serve a different purpose, are of different value to the reader and are actually more important than the document title so they have to come first, as long as we are ok with that for the cluster search it's all good. (in addition to what you have noted above, I think it's possible for each document to have tags too in the cluster search)

In my original designs the tags were not placed at the top for the regular search but they were for cluster search, and in our conversations with you and Nick we questioned the value of the difference between the two layouts so we combined them to one layout.
